src/cdcacm_serial.c

```diff
@@ -95,19 +96,25 @@ static void write_data(struct device *dev, const char *buf, int len)
 
 void cdc_acm_tx()
 {
+	uint32_t dtr = 0;
 	if (acm_tx_state == ACM_TX_READY) 
 	{
 		if(Tx_HEAD != Tx_TAIL)
 		{
-			int cnt = 0;
-			gpio_pin_write(gpio_dev, TXRX_LED, 0);	//turn TXRX led on
+			int cnt = 0;			
 			for (; (Tx_TAIL != Tx_HEAD) && (cnt < BUFFER_LENGTH*2);cnt++)
 			{
 				write_buffer[cnt] = Tx_BUFF[Tx_TAIL];
 				Tx_TAIL = (Tx_TAIL + 1)% SBS;
 			}
-			write_data(dev, (const char*)write_buffer, cnt);
-			gpio_pin_write(gpio_dev, TXRX_LED, 1);	//turn TXRX led off
+			uart_line_ctrl_get(dev, LINE_CTRL_DTR, &dtr);
```

Comment:
Why reading the DTR here? what does it tell you?

trying to figure it out myself....
DTR normally used for flow control right? i.e. receiver on the other end will assert DTR when it's ready to receive more data. So it should be asserted, most of the time.

So, basically you are only doing the delay when DTR is asserted. Shouldn't you do it for every character?